### DISCUSSION TITLE
Info Buffer

### DISCUSSION THREAD

**Discussion by tmattacchione** [liquidz/elin#3] [Q&A]
It's my understanding when `ElinToggleInfoBuffer` is executed it opens a buffer in vim which should contain the evaluated results.  Is this correct?

What I'm currently seeing is a buffer with the text `;; Elin Info Buffer` (as seen below), but when I evaluate using `ElinEvalCurrentList` the Info Buffer doesn't update with the results.

Any guidance would be appreciated.

<img width="1455" alt="Screenshot 2024-11-04 at 7 09 11 PM" src="https://github.com/user-attachments/assets/a0b1b526-3478-4103-8d0c-3f930f57cce2">

**Comment by liquidz** [comment]
@athomasoriginal Evaluation results are not recorded to information buffer by default.

cf. https://liquidz.github.io/elin/#_results

For now, standard outputs are recorded to information buffer.
If you'd like to record the evaluation results, one of the following is needed.

* Execute [ElinPrintLastResult](https://liquidz.github.io/elin/#_elinprintlastresult) command when you needed
* Configure to always record all evaluated results 
  * (There were a bug on evaluation handler, so you need to update elin for this configuration)
  ```clojure
  {:handler {:config-map {elin.handler.evaluate/evaluate-current-expr {:interceptor {:includes [elin.interceptor.handler/append-result-to-info-buffer]}}
                          elin.handler.evaluate/evaluate-current-list {:interceptor {:includes [elin.interceptor.handler/append-result-to-info-buffer]}}
                          elin.handler.evaluate/evaluate-current-top-list {:interceptor {:includes [elin.interceptor.handler/append-result-to-info-buffer]}}}}}
  ```

**Reply by tmattacchione** [reply]
Thanks @liquidz!

* [ElinPrintLastResult](https://liquidz.github.io/elin/#_elinprintlastresult) works :) 
* I set the config and it's picked up by Elin, but  when I run `ElinCurrentList` I get an error in nvim saying, "Failed to Intercept for: :Elin.constant.interceptor/nrepl: Broken pipe".  Not a problem because `ElinPrintLastResult`!

**Reply by liquidz** [reply]
@athomasoriginal 
> Failed to Intercept for: :Elin.constant.interceptor/nrepl: Broken pipe

Hmm, I could not reproduce the problem in my environment.

https://github.com/user-attachments/assets/cfb94b98-0f50-46b9-9600-7618a664a9d2

If possible, could you open an issue as a bug report?
